src/main/java/ifsul/agileproject/rachadinha/service/impl/RachaServiceImpl.java

```diff
@@ -120,16 +159,24 @@ public void addMemberToRacha(Long rachaId, Long userId, String password) {
     }
     Racha racha = rachaRepository.findById(rachaId).get();
     User user = userRepository.findById(userId).get();
-    if (racha.getMembers().contains(user)) {
-			throw new UserAlreadyInRachaException(userId, rachaId);
-		}
+    Payment existingPayment = paymentRepository.findByRachaAndUser(racha, user);
+    if (!racha.getStatus().equals(Status.OPEN)) {
+      throw new JoinRachaForbiddenException(rachaId);
+    }
+    if (existingPayment != null) {
+      throw new UserAlreadyInRachaException(userId, rachaId);
+    }
     if (!racha.getPassword().equals(password)) {
       throw new IncorrectRachaPasswordException(rachaId);
     }
-    racha.getMembers().add(user);
-    user.getRachas().add(racha);
-    rachaRepository.save(racha);
-    userRepository.save(user);
+    Payment payment = Payment.builder()
+        .racha(racha)
+        .user(user)
+        .userSaidPaid(false)
+        .confirmedByOwner(true)
```

Comment:
When the user join into a racha, the owner already confirm?

Reply:
That is how i did it, but we can change 😅
The owner is still going to be able to say if they have already paid. As `confirmedByOwner` is true by default, they will not have to confirm their own `userSaidPaid` state later.
We can add further logic to this, but i think the current implementation simplifies this process to the owner.

Reply:
This behavior should apply only to the owner. Regular users joining the racha should have both `userSaidPaid` and `confirmedByOwner` set as `false`.